.circleci/config.yml

```diff
@@ -0,0 +1,156 @@
+            pip install virtualenv
+            virtualenv env
+            $CIRCLE_WORKING_DIRECTORY/env/bin/pip install --upgrade pip
+            $CIRCLE_WORKING_DIRECTORY/env/bin/pip install -r requirements.txt
+      - run:
+          name: Run tests
+          command: |
+            # JS Unit Tests
+            cd $CIRCLE_WORKING_DIRECTORY/tests/client
+            $CIRCLE_WORKING_DIRECTORY/client/node_modules/.bin/karma start ./karma.conf.js \\
+              --single-run --browsers PhantomJS --reporters junit
+            # Run Python tests
+            cd $CIRCLE_WORKING_DIRECTORY
+            env/bin/nosetests ./tests/server --with-xunit \\
+              --xunit-file $CIRCLE_TEST_REPORTS/unitresults.xml \\
+              --with-coverage --cover-erase --cover-package=./server
+            env/bin/coverage xml -o $CIRCLE_TEST_REPORTS/coverage.xml
+      - save_cache:
+          key: cached-directories
+          paths:
+            - $CIRCLE_WORKING_DIRECTORY/client/node_modules
+            - $CIRCLE_WORKING_DIRECTORY/env
+  deploy-staging:
+    working_directory: /usr/src/app
+    docker:
+      - image: python:3-stretch
+    steps:
+      - checkout
+      - setup_remote_docker
+      - install-deploy-deps
+      - run: 
+          name: Remove last snapshot and backup database
+          command: |
+            # Copy old snapshot to temporary
+            if [ -z $(aws rds describe-db-snapshots --filters Name='db-instance-id',Values='tm3-staging-deployment-snapshot-latest' --output text)]
+            then 
+                echo "Snapshot does not exist, creating one now."
+            else
+                aws rds copy-db-snapshot --source-db-snapshot tm3-staging-deployment-snapshot-latest --target-db-snapshot tm3-staging-deployment-snapshot-temp
+            fi
+            # create new aws rds snapshot
+            aws rds create-db-snapshot --db-snapshot-identifier tm3-staging-deployment-snapshot-latest --db-instance-identifier tasking-manager-staging 
+      - run: 
+          name: Create config file
+          command: |
+            printf '{\"GitSha\":\"$CIRCLE_SHA1\", \"Environment\":\"staging\", \"DBSnapshot\":\"$DB_SNAPSHOT_STAGING\", \"MasterUsername\":\"$MASTER_USERNAME_STAGING\", \"MasterPassword\":\"$MASTER_PASSWORD_STAGING\", \"OSMConsumerKey\":\"$OSM_CONSUMER_KEY_STAGING\",\"OSMConsumerSecret\":\"$OSM_CONSUMER_SECRET_STAGING\", \"TaskingManagerSecret\":\"$TASKINGMANAGER_SECRET_STAGING\", \"TaskingManagerEnv\":\"$TASKINGMANAGER_ENV_STAGING\", \"TaskingManagerSMTPHost\":\"$TASKINGMANAGER_SMTP_HOST_STAGING\", \"TaskingManagerSMTPPassword\":\"$TASKINGMANAGER_SMTP_PASSWORD_STAGING\", \"TaskingManagerSMTPUser\":\"$TASKINGMANAGER_SMTP_USER_STAGING\", \"DatabaseSize\":\"$DATABASE_SIZE\",\"ELBSubnets\":\"$ELB_SUBNETS_STAGING\", \"SSLCertificateIdentifier\":\"$SSL_CERTIFICATE_ID_STAGING\",\"RDSUrl\":\"$RDS_URL_STAGING\"}' > $CIRCLE_WORKING_DIRECTORY/cfn-config-staging.json
+      - aws-s3/copy:
```

Comment:
Have you checked the contents of the file on S3? Does it look like a JSON string (`\"<string>\"`) or a JSON ("<string>"). We want the latter.

Reply:
Those `\"` should be parsed by `printf` into regular `"`